### PR TITLE
Add device doctor license and readme

### PR DESCRIPTION
**Pull request by keyonghan**
This adds LICENSE and README to device_doctor tool.

Related issue: https://github.com/flutter/flutter/issues/72835

### CODE REVIEW

File: device_doctor/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# device_doctor
+
+This utility tool is used by LUCI infrastructure to manage device health
+for Flutter LUCI swarming bots.
+
+It offers support for different platforms: linux, mac and windows, and
```
Comment (CaseyHillers) [inline device_doctor/README.md:6]: platforms->host platforms is clearer as Android and iOS are platforms too.

File: device_doctor/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+### iOS
+* `idevice_id`
+* `idevicediagnostics`
+* `xcrun`
+
+## Building
+
+This tool is meant to be published as an AOT compiled binary distributed via
+CIPD. To build the AOT binary, run `tool/build.sh` for linux and mac, and run
```
Comment (CaseyHillers) [inline device_doctor/README.md:24]: Should we instead redirect users to get the latest version from CIPD? There's already a manual build section below

File: device_doctor/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+cipd create -in build                   \
+  -name flutter/device_doctor/<os>-amd64 \
+  -ref <ref>                     \
+  -tag sha_timestamp:<revision>_<timestamp>
+```
+
+* os: `linux`, `mac`, or `windows`.
+* ref: `release` or `staging`
+
+## How to use
```
Comment (CaseyHillers) [inline device_doctor/README.md:55]: If `device_doctor` has a help message, we should list that here instead. Commands and flags tend to get out of date from the documentation.

**Review by CaseyHillers — APPROVED** [review]
LGTM w/ comments

File: device_doctor/README.md (hunk 1 above)
Comment (keyonghan) [reply device_doctor/README.md:6]: Done.

File: device_doctor/README.md (hunk 2 above)
Comment (keyonghan) [reply device_doctor/README.md]: Updated.

File: device_doctor/README.md (hunk 3 above)
Comment (keyonghan) [reply device_doctor/README.md:55]: used `help` instead, and updated the args to make it more clear.

**Pull request merged**